### PR TITLE
Payload command behavior

### PR DESCRIPTION
**Pull request by patapufate**
# Description
This PR adds the Payload Command behavior. This behavior publishes a command passed as argument (BT::port) at the behavior/action 
The behavior description is defined:
```
<PayloadCommandBehavior     command="command"/>
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing

### CODE REVIEW

File: mission_control/include/mission_control/behaviors/payload_command.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, QinetiQ, Inc.
```
Comment (hidmic) [inline mission_control/include/mission_control/behaviors/payload_command.h:4]: @patapufate meta: should read 2021? Anyhow, we can fix the entire `experimental` branch right before merging into  `master`

File: mission_control/src/behaviors/payload_command.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  {
+    publishPayloadCommandMsg();
+    payloadCommandHasBeenPublished_ = true;
+  }
+  else
+  {
+    setStatus(BT::NodeStatus::SUCCESS);
+    payloadCommandHasBeenPublished_ = false;
+  }
+  return (status());
```
Comment (hidmic) [inline mission_control/src/behaviors/payload_command.cpp:65]: @patapufate does the payload manager provide any feedback? If not, why not publishing and succeeding immediately (as opposed to this two step approach)?

File: mission_control/include/mission_control/behaviors/payload_command.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+{
+ public:
+  PayloadCommandBehavior(const std::string& name, const BT::NodeConfiguration& config);
+
+  BT::NodeStatus behaviorRunningProcess();
+
+  static BT::PortsList providedPorts()
+  {
+    BT::PortsList ports = {BT::InputPort<std::string>("command", " ", "command")};
+    return ports;
```
Comment (hidmic) [inline mission_control/include/mission_control/behaviors/payload_command.h:61]: @patapufate meta: should ports be `static` as well?

File: mission_control/include/mission_control/behaviors/payload_command.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+class PayloadCommandBehavior : public Behavior
+{
+ public:
+  PayloadCommandBehavior(const std::string& name, const BT::NodeConfiguration& config);
+
+  BT::NodeStatus behaviorRunningProcess();
+
+  static BT::PortsList providedPorts()
+  {
+    BT::PortsList ports = {BT::InputPort<std::string>("command", " ", "command")};
```
Comment (hidmic) [inline mission_control/include/mission_control/behaviors/payload_command.h:60]: @patapufate why an empty a default value? To fail on behavior construction if `command` was not specified sounds best.

File: mission_control/include/mission_control/behaviors/payload_command.h (hunk 1 above)
Comment (patapufate) [reply mission_control/include/mission_control/behaviors/payload_command.h:4]: Fixed.

File: mission_control/src/behaviors/payload_command.cpp (hunk 2 above)
Comment (patapufate) [reply mission_control/src/behaviors/payload_command.cpp:65]: Yes you're right. I left the two-step approach to future implementations, but, sounds reasonable your approach

File: mission_control/include/mission_control/behaviors/payload_command.h (hunk 4 above)
Comment (patapufate) [reply mission_control/include/mission_control/behaviors/payload_command.h:60]: Fixed

**Review by hidmic — COMMENTED** [review]
@patapufate you sure you pushed all changes?

**Review by hidmic — APPROVED** [review]
LGTM !

**Pull request merged**